### PR TITLE
[IMP] web: update accepts commands for x2m

### PR DESCRIPTION
**Pull request by FrancoisGe**
Before this commit, the record.update function did not allow x2m fields
to be updated. Thanks to this commit, you can update an x2m by passing
a list of commands that will be applied to the x2m's static list.

This makes it possible to update several fields, including x2m fields,
while triggering only one onchange to the server.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by chevalierv** [comment]
I look like it is a work in progress. I put the PR in draft for the moment

File: addons/survey/static/src/question_page/question_page_one2many_field.js [hunk 1]
```diff
@@ -100,12 +100,13 @@ class QuestionPageOneToManyField extends X2ManyField {
                 return;
             }
             if (params.record) {
+                params.record = self.props.record.data[self.props.name].records.find(r => r.isNew ? r.virtualId === params.record.virtualId : r.resId === params.record.resId)
```
Comment (FrancoisGe) [inline addons/survey/static/src/question_page/question_page_one2many_field.js:103]: rem isVirtual

File: addons/sale_product_configurator/static/src/js/sale_product_field.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 /** @odoo-module */
 
-import { patch } from "@web/core/utils/patch";
-import { useService } from "@web/core/utils/hooks";
 import { SaleOrderLineProductField } from '@sale/js/sale_product_field';
 import { serializeDateTime } from "@web/core/l10n/dates";
+import { x2ManyCommands } from "@web/core/orm_service";
+import { useService } from "@web/core/utils/hooks";
+import { patch } from "@web/core/utils/patch";
```
Comment (chevalierv) [inline addons/sale_product_configurator/static/src/js/sale_product_field.js:7]: Hello :wave: Why are those import needed ? :)

**Review by chevalierv — APPROVED** [review]

**Review by aab-odoo — APPROVED** [review]
robodoo r+
